core/templates/dev/head/domain/exploration/OutcomeObjectFactory.js

```diff
@@ -35,6 +35,18 @@ oppia.factory('OutcomeObjectFactory', [
       };
     };
 
+    /** Returns true iff an outcome has a self-loop and no feedback. */
+    Outcome.prototype.isConfusing = function(currentStateName) {
+      return (
+        this.dest === currentStateName &&
+        !this.feedback.getHtml().trim().length > 0
```

Comment:
Perhaps this should use hasNonemptyFeedback() -- and the feedback.isEmpty() check should include a trim() as well if it doesn't already.

Reply:
done